### PR TITLE
Feature/incoming token transactions

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

https://trello.com/c/GbmznAVK/104-view-received-tokens

Needed to work https://github.com/MetaMask/controllers/pull/247

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/util/transactions.js [hunk 1]
```diff
@@ -264,7 +279,16 @@ export async function getTransactionActionKey(transaction) {
  * @returns {string} - Transaction type message
  */
 export async function getActionKey(tx, selectedAddress, ticker, paymentChannelTransaction) {
+	if (tx.isTransfer) {
+		const selfSent = safeToChecksumAddress(tx.transaction.from) === selectedAddress;
+		if (selfSent) {
+			return strings('transactions.self_sent_unit', { unit: tx.transferInformation.symbol });
+		}
+		return strings('transactions.received_unit', { unit: tx.transferInformation.symbol });
```
Comment (rickycodes) [inline app/util/transactions.js:287]: Maybe we can simplify this slightly? Something like:
```suggestion
		return strings(selfSent ? 'transactions.self_sent_unit' : 'transactions.received_unit', { unit: tx.transferInformation.symbol });
		
```
If that's not clear enough maybe:
```suggestion
		const translationKey = selfSent ? 'transactions.self_sent_unit' : 'transactions.received_unit';
		return strings(translationKey, { unit: tx.transferInformation.symbol });
		
```

File: app/components/UI/TransactionElement/utils.js [hunk 2]
```diff
@@ -163,10 +178,16 @@ function getTokenTransfer(args) {
 		};
 	}
 
+	let transactionType;
+	if (renderFullAddress(from) === selectedAddress) transactionType = TRANSACTION_TYPES.SENT_TOKEN;
+	else transactionType = TRANSACTION_TYPES.RECEIVED_TOKEN;
+
```
Comment (rickycodes) [inline app/components/UI/TransactionElement/utils.js:184]: I think using let and if/else conditionals makes sense if there's greater than two possibilities, but for cases like this we can use a ternary and `const`:

(I've also added some destructuring for the `TRANSACTION_TYPES`, but that can likely go at the `import`)

```suggestion
	const { SENT_TOKEN, RECEIVED_TOKEN } = TRANSACTION_TYPES
	const transactionType = (renderFullAddress(from) === selectedAddress)
		? SENT_TOKEN
		: RECEIVED_TOKEN
```

File: app/components/UI/TransactionElement/utils.js [hunk 3]
```diff
@@ -223,10 +245,111 @@ function getCollectibleTransfer(args) {
...
+		? `${renderFromTokenMinimalUnit(amount, token.decimals)} ${token.symbol}`
+		: strings('transaction.value_not_available');
+	const totalFiatNumber = renderTokenFiatNumber
+		? weiToFiatNumber(totalGas, conversionRate) + renderTokenFiatNumber
+		: weiToFiatNumber(totalGas, conversionRate);
+
+	const ticker = getTicker(args.ticker);
+	let transactionType;
+	if (renderFullAddress(from) === selectedAddress) transactionType = TRANSACTION_TYPES.SENT_TOKEN;
+	else transactionType = TRANSACTION_TYPES.RECEIVED_TOKEN;
```
Comment (rickycodes) [inline app/components/UI/TransactionElement/utils.js:311]: same as above :point_up:

File: app/components/UI/TransactionElement/utils.js [hunk 4]
```diff
@@ -298,6 +427,10 @@ function decodeTransferFromTx(args) {
 	const renderTo = renderFullAddress(addressTo);
 	const ticker = getTicker(args.ticker);
 
+	let transactionType;
+	if (renderFrom === selectedAddress) transactionType = TRANSACTION_TYPES.SENT_COLLECTIBLE;
+	else transactionType = TRANSACTION_TYPES.RECEIVED_COLLECTIBLE;
+
```
Comment (rickycodes) [inline app/components/UI/TransactionElement/utils.js:433]: same as above :point_up:

**Review by rickycodes — APPROVED** [review]
Mostly looks good to me, just a few small notes.

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
**Issue 1:** 

The txn history updates roughly every 15 seconds or so = http://recordit.co/iH31hzRKLQ 
If you are viewing txn details and have the modal open, this action will automatically close the modal = http://recordit.co/HnTwgPoYj1

**Issue 2:**

I wanted to test canceling a txn, but found that I can't send any ETH on Rinkeby or Ropsten, this was the error I got on Rinkeby

image

Followed by this:
Screen Shot 2020-07-08 at 8 04 26 PM

**Issue 3:**

Padding is a bit off on the icon's on Android; seen here on a pixel 2 device

image

**Comment by estebanmino** [comment]
Issue 1 fixed with https://github.com/MetaMask/controllers/pull/247/commits/0db69d911300276eb216dc57f6ff27527ee0e7ee
The other two on this PR

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
Issues 1 -3 have been resolved 👍 

**Issue 4:**

Something weird with SAI is occurring where I sent SAI to an account, but it's appearing as received DAI; seen here = https://recordit.co/q5S2nZ0Hj5

image

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good, QA Passed 👍

**Pull request merged**